cargo-guppy/src/main.rs

```diff
@@ -26,6 +27,37 @@ enum Command {
     #[structopt(name = "dups")]
     /// Print the number of duplicate packages
     Duplicates,
+    #[structopt(name = "select")]
+    /// Select packages and their transitive dependencies
+    Select {
+        #[structopt(long, short, possible_values = &Kind::variants(), case_insensitive = true, default_value = "all")]
+        /// Kind of crates to select
+        kind: Kind,
+        #[structopt(long, rename_all = "kebab-case")]
+        /// Include dev dependencies
+        include_dev: bool,
+        #[structopt(long, rename_all = "kebab-case")]
+        /// Include build dependencies
+        include_build: bool,
+        #[structopt(long)]
+        /// Target to select for, default is to match all targets
+        target: Option<String>,
+        #[structopt(long)]
+        /// Save selection graph in .dot format
+        output: Option<String>,
+        #[structopt(
+            long,
+            rename_all = "kebab-case",
+            name = "package",
+            number_of_values = 1
+        )]
+        /// Omit edges that point into a given package; useful for seeing how
+        /// removing a dependency affects the graph
+        omit_edges_into: Vec<String>,
+        #[structopt(rename_all = "screaming_snake_case")]
+        /// The root packages to start the selection from
+        roots: Vec<String>,
```

Comment:
Is there any way these params can be expressed as a struct so you don't have to pass so many parameters into `cmd_select`?

Reply:
Maybe https://docs.rs/structopt/0.3.5/structopt/#flattening ?

Reply:
I forgot to reorganize this when I was done. I'll fix this.